Shared/Flows/Onboarding/Welcome/WelcomeViewController.swift

```diff
@@ -78,19 +79,40 @@ class WelcomeViewController: DiffableCollectionViewController<MessageSequenceSec
             self.didLoadConversation?(convo)
         }
         
-        Task {
-            await Task.sleep(seconds: 0.2)
-            self.welcomeCollectionView.timeMachineLayout.prepare()
-            let maxOffset = self.welcomeCollectionView.timeMachineLayout.maxZPosition
-            self.collectionView.setContentOffset(CGPoint(x: 0, y: maxOffset), animated: false)
-            self.welcomeCollectionView.timeMachineLayout.invalidateLayout()
-        }.add(to: self.autocancelTaskPool)
+        self.updateContentOffset()
     }
     
     override func getAnimationCycle(with snapshot: NSDiffableDataSourceSnapshot<MessageSequenceSection, MessageSequenceItem>) -> AnimationCycle? {
-        return AnimationCycle(inFromPosition: .inward,
-                              outToPosition: .inward,
-                              shouldConcatenate: true)
+        let count = (snapshot.numberOfItems(inSection: .topMessages) + snapshot.numberOfItems(inSection: .bottomMessages)) - 1
+        let maxOffset = CGFloat(count) * self.welcomeCollectionView.timeMachineLayout.itemHeight
+        
+        return AnimationCycle(inFromPosition: nil,
+                              outToPosition: nil,
+                              shouldConcatenate: false,
+                              scrollToOffset: CGPoint(x: 0, y: maxOffset))
+    }
+    
+    var scrollTask: Task<Void, Never>?
+    func updateContentOffset() {
+        // Cancel any currently running swipe hint tasks so we don't trigger the animation multiple times.
+        self.scrollTask?.cancel()
+        
+        self.scrollTask = Task {
```

Comment:
Looks like this will scroll forever and cause a memory leak. Use weak self to prevent this.

Or cancel the task in view will disappear and make sure to guard for cancel after the last snooze.